cli/src/cli_util.rs

```diff
@@ -1012,7 +1012,25 @@ impl WorkspaceCommandHelper {
         let immutable = revset_util::parse_immutable_expression(&self.revset_parse_context())?;
         let mut expression = self.attach_revset_evaluator(immutable)?;
         expression.intersect_with(&to_rewrite_revset);
-        if let Some(commit_id) = expression.evaluate_to_commit_ids()?.next() {
+        if let Some(commit_id) = expression
+            .evaluate_to_commit_ids()
+            .map_err(|e| {
+                let UserRevsetEvaluationError::Resolution(RevsetResolutionError::NoSuchRevision {
+                    name,
+                    candidates: _,
+                }) = e
+                else {
+                    return e.into();
+                };
+                user_error(format!(
+                    "The expression '{}' of immutable_heads() could not be resolved, if you \
+                     configured specific branch names consider wrapping them with 'present()' or \
+                     using the functions 'branches()' / 'remote_branches()'.",
```

Comment:
Use a raw string here (`r#"foo"#`) and inline the identifier.

Reply:
I meant 
```rust 
format!(r#""The expression '{name}' of immutable_heads() ..."#);
```

Reply:
wanted to ask what you meant by that but had to do something else, makes sense